### ISSUE TITLE
Is kube-rbac-proxy required?

### ISSUE THREAD

**Issue by micahhausler** [kubernetes-sigs/cluster-api#4325]
**User Story**

As a [distributor of Cluster-API] I would like to [distribute container images for CAPI] for [reasons].

**Detailed Description**

[kube-rbac-proxy](https://github.com/brancz/kube-rbac-proxy) seems like a great project, and this is not meant to imply any mistrust of @brancz or his work, but distributing a security proxy that is a developer's personal project is concerning from both a a supply-chain and governance point of view. 

Can someone clarify if kube-rbac-proxy is a required sidecar for CAPI controllers? If it is not, can templates and documentation be updated to reflect that and provide installation instructions that don't require it?

**Anything else you would like to add:**

My initial read is that kube-rbac-proxy is just being used to guard `/metrics` endpoints for Prometheus, but is there more to it then that? 

/kind feature
/kind documentation

**Comment by brancz** [comment]
I don't know about cluster-api things, but FWIW I've attempted to donate kube-rbac-proxy to sig-auth, and there was interest from sig-auth to adopt it, I just don't have the time to make this happen right now. I think @s-urbaniak wanted to continue this.

**Comment by s-urbaniak** [comment]
Indeed yes, I would love to help in contributing kube-rbac-proxy to sig-auth! We use it in various places and maintain it continuously.

@micahhausler quick question: what are CAPI controllers? To answer your question: generally, kube-rbac-proxy can be used in front of any endpoint that requires authentication/authorization based on kubernetes tokens. Internally, it uses token reviews (optionally delegating to oidc servers) and subject access reviews.

@deads2k (cc @stlaz) do you, by any chance have any additional pointers how we can proceed with sig-auth donation?

**Comment by vincepri** [comment]
Hi folks! Trying to connect this conversation to other individuals. 

FWIW, we're using kube-rbac-proxy because Cluster API is a project based on Kubebuilder.

**Comment by vincepri** [comment]
@brancz Would you be able to send an email to both SIG Auth and SIG API Machinery mailing lists to ask to adopt the project under the kubernetes-sigs org? We should make it clear that this project is used in all of Kubebuilder projects, and it has so far provided great value to the community. In addition, having it under the kubernetes-sigs can have the good side effect of increasing contribution and adoption even more.

**Comment by s-urbaniak** [comment]
@brancz willing to send out the email from my side just to offload you unless it is necessary for the original maintainer to do so.

**Comment by micahhausler** [comment]
kube-rbac-proxy seems like a great k8s-sigs addition. From assisting with other projects in the past, here are [the steps for external projects to be donated](https://github.com/kubernetes/community/blob/master/github-management/kubernetes-repositories.md#rules-for-donated-repositories)

Can I get an answer to this original question?

> My initial read is that kube-rbac-proxy is just being used to guard /metrics endpoints for Prometheus, but is there more to it then that?

**Comment by s-urbaniak** [comment]
@micahhausler yes, as mentioned above it can be used in any context where token based authentication/authorization is necessary inside the cluster. For instance, we use it (in OpenShift) as well to protect user access to the Prometheus UI, the Alertmanager UI, Thanos UI, etc. kube-rbac-proxy is dynamic in its configuration against which resources you want to have protection against and we plan to make a local static authorizer (to reduce API server load) as well.

Does that answer your question?

**Comment by micahhausler** [comment]
> Does that answer your question?

Not quite. I understand how it works in the general sense, but I'm asking _specifically_ about cluster-api. Are there non-metrics endpoints being protected, or is it exclusively used for metrics protection?

**Comment by s-urbaniak** [comment]
Ah sorry, I am not involved in cluster-api, so maybe others can chime in.

**Comment by vincepri** [comment]
@micahhausler I don't think so, but we need to double check. This might warrant a little poc to see what would break.

**Comment by brancz** [comment]
I'm not much involved in kube-rbac-proxy at the moment, but @s-urbaniak has full ownership on the repo so I'm happy with him taking the lead on the donation of it to sig-auth.

**Comment by randomvariable** [comment]
I'm fairly confident that it's only being used for metrics.

**Comment by sbueringer** [comment]
> I'm fairly confident that it's only being used for metrics.

Me too. But we'll see in the PR. The only other endpoints we have are the webhooks and they are not using kube-rbac-proxy.
